policy-controller/k8s/index/src/index.rs

```diff
@@ -1070,13 +1082,58 @@ impl Pod {
             );
         };
 
+        // No Server is configured, so requests on the Pod's probe paths are
+        // authorized.
+        let http_routes = {
+            if config.probe_networks.is_empty() {
```

Comment:
we never want probe routes when the probe networks are empty, right?  so I think it makes sense to push this logic down into http_probe_routes and have that function return HashMap::default() when networks is_empty().